In [37]:
import pandas as pd
import numpy as np

In [38]:
data = pd.read_csv("./users.data", sep ="\t",names=["user_id", "item_id","rating","timestamp"])
data

user_id  item_id  rating  timestamp
0             0       50       5  881250949
1             0      172       5  881250949
2             0      133       1  881250949
3           196      242       3  881250949
4           186      302       3  891717742
...         ...      ...     ...        ...
99998       880      476       3  880175444
99999       716      204       5  879795543
100000      276     1090       1  874795795
100001       13      225       2  882399156
100002       12      203       3  879959583

[100003 rows x 4 columns]

### **VERİ OKUMA İŞLEMİNDE DÜZENLEME YAPILMAZSA VERİ BU ŞEKİLDE OLUR**

-	0\t50\t5\t881250949
- 0	0\t172\t5\t881250949
- 1	0\t133\t1\t881250949
- 2	196\t242\t3\t881250949
- 3	186\t302\t3\t891717742
- 4	22\t377\t1\t878887116
- ...	...
- 99997	880\t476\t3\t880175444
- 99998	716\t204\t5\t879795543
- 99999	276\t1090\t1\t874795795
- 100000	13\t225\t2\t882399156
- 100001	12\t203\t3\t879959583
- 100002 rows × 1 columns

In [39]:
data.isnull().sum()

user_id      0
item_id      0
rating       0
timestamp    0
dtype: int64

In [40]:
movies = pd.read_csv("./movie_id_titles.csv")
movies

item_id                                      title
0           1                           Toy Story (1995)
1           2                           GoldenEye (1995)
2           3                          Four Rooms (1995)
3           4                          Get Shorty (1995)
4           5                             Copycat (1995)
...       ...                                        ...
1677     1678                          Mat' i syn (1997)
1678     1679                           B. Monkey (1998)
1679     1680                       Sliding Doors (1998)
1680     1681                        You So Crazy (1994)
1681     1682  Scream of Stone (Schrei aus Stein) (1991)

[1682 rows x 2 columns]

In [41]:
movies.isnull().sum()

item_id    0
title      0
dtype: int64

In [42]:
# movies DataFrame'in title sütununu, data DataFrame'ine eklemek için merge işlemi yapıyoruz
data = pd.merge(data, movies[['item_id', 'title']], on="item_id")

# Yeni data DataFrame'inin ilk birkaç satırını görmek için
data.head()


user_id  item_id  rating  timestamp                            title
0        0       50       5  881250949                 Star Wars (1977)
1        0      172       5  881250949  Empire Strikes Back, The (1980)
2        0      133       1  881250949        Gone with the Wind (1939)
3      196      242       3  881250949                     Kolya (1996)
4      186      302       3  891717742         L.A. Confidential (1997)

In [43]:
# Yeni DataFrame'i CSV dosyasına yazdırıyoruz
data.to_csv("merge_data.csv", index=False)